frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -269,8 +269,14 @@ function QueryBuilder(props) {
 
   const handleCreate = useCallback(
     async card => {
-      const questionWithUpdatedCard = question.setCard(card).setPinned(false);
+      const shouldBePinned = Boolean(card.dataset);
```

Comment:
```suggestion
      const shouldBePinned = !!card.dataset;
```

This is better because you get to say "bang bang" when you read it. 😁

Reply:
And resort to some animalistic onomatopoeia to the detriment of a much called-for homage to the immortal George Boole? Never!

Reply:
🤣